kernel/src/expressions/column_names.rs

```diff
@@ -572,4 +637,45 @@ mod test {
             assert_eq!(parsed.to_string(), expected_output);
         }
     }
+
+    #[test]
+    fn test_parse_column_name_list() {
+        let cases = [
+            ("", Some(vec![])),
+            (
+                "  ,  ",
+                Some(vec![ColumnName::empty(), ColumnName::empty()]),
+            ),
+            ("  a  ", Some(vec![column_name!("a")])),
+            (
+                "  ,  a  ",
+                Some(vec![ColumnName::empty(), column_name!("a")]),
+            ),
+            (
+                "  a  ,  ",
+                Some(vec![column_name!("a"), ColumnName::empty()]),
+            ),
```

Comment:
hm... thinking about this. do we want trailing delimiters to mean an empty column follows? I feel like SQL has annoying behavior like this but I've gotten used to trailing delimiters being optional (e.g. in rust syntax)

Reply:
I thought about that, but we already allow a whole bunch of similar corner cases already, such as `"."` parsing as a column name with two empty field names. If an engine cares doesn't like those, it's welcome to check for and reject them.

Reply:
(if we decide to reject any of the corner cases, we probably need to start rejecting them all, which is a lot of extra code and testing to worry about -- but we can do it as a separate PR if we decide it's really needed)

Reply:
sounds good :)